dali/pipeline/operators/reader/nvdecoder/cuvideodecoder.cc

```diff
@@ -103,19 +113,70 @@ CUVideoDecoder& CUVideoDecoder::operator=(CUVideoDecoder&& other) {
     }
     decoder_ = other.decoder_;
     initialized_ = other.initialized_;
+    max_height_ = other.max_height_;
+    max_width_ = other.max_width_;
     other.decoder_ = 0;
     other.initialized_ = false;
+    other.max_height_ = 0;
+    other.max_width_ = 0;
     return *this;
 }
 
+void CUVideoDecoder::reconfigure(unsigned int height, unsigned int width) {
+    DALI_ENFORCE(NVCUVID_API_EXISTS(cuvidReconfigureDecoder),
+                 "cuvidReconfigureDecoder API is not available.");
+
+    CUVIDRECONFIGUREDECODERINFO reconfigParams = { 0 };
+
+    DALI_ENFORCE(initialized_, "Trying to reconfigure uninitialized decoder");
+
+    if (width < caps_.nMinWidth ||
+        height < caps_.nMinHeight) {
+        DALI_FAIL("Video is too small in at least one dimension.");
```

Comment:
Can we output the exact file name here so the user knows what is causing this troubles?

Reply:
Added here.
https://github.com/NVIDIA/DALI/blob/3adac83832bb813e5870fd40a55a367352a3859b/dali/pipeline/operators/reader/nvdecoder/nvdecoder.cc#L150-L154

Reply:
... shouldn't it rethrown so python can catch this?

Reply:
It's rethrown from main thread

https://github.com/NVIDIA/DALI/blob/3adac83832bb813e5870fd40a55a367352a3859b/dali/pipeline/operators/reader/nvdecoder/nvdecoder.cc#L357-L358

https://github.com/NVIDIA/DALI/blob/3adac83832bb813e5870fd40a55a367352a3859b/dali/pipeline/operators/reader/nvdecoder/nvdecoder.cc#L152-L157